context/Chat.tsx

```diff
@@ -0,0 +1,183 @@
+import { CustomEvent, SpaceEvent } from "@mux/spaces-web";
+import { useSpaceEvent } from "hooks/useSpaceEvent";
+import React, {
+  createContext,
+  ReactNode,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import SpaceContext from "./Space";
+import { v4 } from "uuid";
+import moment from "moment";
+
+const NUM_CHAT_MESSAGES = 100;
```

Comment:
Why set a limit?

Reply:
I wanted to bound the number of messages to something reasonable instead of letting it be unbounded, which could potentially slow down the browser.

Reply:
I don't think we should limit this unless we experience performance issues ourselves. At the very least we should bump the limit to something like 5,000 messages. But, I feel confident we could go much higher which is why I don't think we need a limit.

Also, a really good solution for potentially large lists of data is to use [React Virtualized](https://github.com/bvaughn/react-virtualized) which only renders what is visible on the screen and swaps out DOM nodes as you scroll. Which means you can handle extremely large lists. I ran [this demo](https://bvaughn.github.io/react-virtualized/#/components/List) with 100,000 rows and it didn't break a sweat 😎

Reply:
Sounds good, I've gone ahead and removed the limit as I didn't run into any issues myself.